lib/temporal/workflow/context.rb

```diff
@@ -123,27 +124,43 @@ def execute_workflow(workflow_class, *input, **args)
         )
 
         target, cancelation_id = schedule_command(command)
-        future = Future.new(target, self, cancelation_id: cancelation_id)
+
+        # child_workflow_execution_future tracks when the workflow execution starts
+        child_workflow_execution_future = Future.new(target, self, cancelation_id: cancelation_id)
+
+        child_workflow_future = ChildWorkflowFuture.new(
```

Comment:
nit: you kinda already pass all the information need to create `child_workflow_execution_future` into the `ChildWorkflowFuture`, so you might avoid changing the initializer's signature and create the `Future` from within the `ChildWorkflowFuture`

Reply:
nice idea, done!